### ISSUE TITLE
Wish: Place new nodes into unoccupied zones of the canvas

### ISSUE THREAD

**Issue by andydandy74** [DynamoDS/Dynamo#3701]
Today I was placing a couple of new nodes and was surprised that I couldn't find them on canvas. It truned out they were hidden behind a large text note. It would be great if Dynamo could check the placement coordinates of new nodes against existing objects on the canvas.

**Comment by tatlin** [comment]
would drag and drop be a better way to handle this?

On Thu, Feb 5, 2015 at 10:02 AM, Andreas Dieckmann <notifications@github.com

**Comment by andydandy74** [comment]
drag and drop would be an alternative, yes. what would be really nice, though, is the ability to create new nodes in canvas (bring up a search field in canvas next to the mouse, enter search string, select one of the results and have it placed at your mouse location).

**Comment by tatlin** [comment]
Both of which existed in previous versions of dynamo!

-matt

**Comment by elayabharath** [comment]
Noted about this - drag and drop + [incavas search](https://docs.google.com/a/adsk-oss.com/document/d/1TvLrSEeqKAhRBckThe4XbQZBPtwxjtHPPR2b-dzUS1U/edit#bookmark=id.diji5x5bblr6) for node placement are in the process of being built. Other idea [here](https://docs.google.com/a/adsk-oss.com/document/d/1GY4Fmx6DBeY_oD52erWOn4CBE5yGatHW3n3cfuFcGq4/edit#bookmark=id.i32pnexybyyx). Let me know what you think.

**Comment by andydandy74** [comment]
@elayabharath  - can't access that document... :-(

**Comment by elayabharath** [comment]
Sorry, shared now. You should be able to view and comment in the doc, thanks

**Comment by andydandy74** [comment]
@elayabharath - Interesting idea, the halo. Not sure I like that green outline, though - there's no green anywhere else in Dynamo, is there?
Read through the rest of the document as well, lots of exciting ideas in there. I added a couple of comments where I felt it was appropriate.

**Comment by dimven** [comment]
@elayabharath Amazing stuff there. Some of the changes are quite radical and definitely improve things for the better. My only concern is that beginners might be overwhelmed if all the changes are introduced at the same time. 

On a semi-related note, lately when you copy and paste a node, it pastes exactly on top the previous node. Previously the newly pasted node got offset a bit. I ended up with 3-4 copies on top of each other before I figured it out. Not sure if I would call this a problem tho...
